sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasStateServer.scala

```diff
@@ -758,3 +760,12 @@ case class MapStateInfo(
     keySerializer: ExpressionEncoder.Serializer[Row],
     valueDeserializer: ExpressionEncoder.Deserializer[Row],
     valueSerializer: ExpressionEncoder.Serializer[Row])
+
+object TransformWithStateInPandasStateServer {
+  @volatile private var id = 0
```

Comment:
IIUC, `TransformWithStateInPandasPythonRunner` is instantiated per executor so they are in different JVMs, and `TransformWithStateInPandasStateServer` is initialized from each `TransformWithStateInPandasPythonRunner` once. What could be the scenario for handling multiple threads here?

Reply:
There's no much difference comparing to TCP socket. For UDS, we create socket files for each server thread connection, and the files are named with different `serverId` which got updated incrementally. At the end we do cleanup to delete these socket files.